pygpu/basic.py

```diff
@@ -0,0 +1,53 @@
+from string import Template
+from .gpuarray import GpuArray, GpuKernel
+
+
+def _generate_kernel(ctx, cols, upper=True):
+    tmpl = Template("""
+    KERNEL void extract_tri(ga_float *a, ga_uint N) {
+        unsigned int idx = blockIdx.y*blockDim.x*gridDim.x+
+                           blockIdx.x*blockDim.x+threadIdx.x;
```

Comment:
Use the cluda defines here:
threadIdx -> LID
blockDim -> LDIM
blockIdx -> GID
gridDim -> GDIM
.x -> _0
.y -> _1